src/core/auth/qgsauthmanager.cpp

```diff
@@ -398,6 +413,42 @@ const QString QgsAuthManager::disabledMessage() const
   return tr( "Authentication system is DISABLED:\n%1" ).arg( mAuthDisabledMessage );
 }
 
+bool QgsAuthManager::createAndStoreRandomMasterPasswordInKeyChain()
+{
+  QMutexLocker locker( mMasterPasswordMutex.get() );
+  if ( isDisabled() )
+    return false;
+
+  if ( mScheduledDbErase )
+    return false;
+
+  if ( !passwordHelperEnabled() )
+    return false;
+
+  if ( !mMasterPass.isEmpty() )
+  {
+    QgsDebugError( QStringLiteral( "Master password is already set!" ) );
+    return false;
+  }
+
+  const QString newPassword = generatePassword();
+  if ( passwordHelperWrite( newPassword ) )
+  {
+    mMasterPass = newPassword;
+  }
+  else
+  {
+    emit passwordHelperMessageLog( tr( "Master password could not be written to your %1" ).arg( AUTH_PASSWORD_HELPER_DISPLAY_NAME ), authManTag(), Qgis::MessageLevel::Warning );
+    return false;
+  }
+
+  if ( !verifyMasterPassword() )
```

Comment:
I can't think of any occasion when this would happen in practice, but if it does happen the user should probably be notified about this error condition.

Reply:
Ok, added